std/src/main/scala/algebra/std/array.scala

```diff
@@ -0,0 +1,36 @@
+package algebra
+package std
+
+import scala.{specialized ⇒ sp}
+
+package object array extends ArrayInstances
+
+trait ArrayInstances {
+  implicit def arrayOrder[@sp A: Order]: Order[Array[A]] = new ArrayOrder[A]
+}
+
+private object ArraySupport {
+
+  def eqv[@sp A: Eq](x: Array[A], y: Array[A]): Boolean = {
+    var i = 0
```

Comment:
is there any benefit to store `x.length` in a local val, or with the jit totally take care of that?